digitalocean/spaces/datasource_spaces_key.go

```diff
@@ -0,0 +1,68 @@
+package spaces
+
+import (
+	"context"
+	"log"
+
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/config"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceDigitalOceanSpacesKey() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSpacesKeyRead,
+
+		Schema: spacesKeySchema(),
+	}
+}
+
+func dataSourceSpacesKeyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	opts := &godo.ListOptions{
+		Page:    1,
+		PerPage: 200,
+	}
+
+	var key *godo.SpacesKey
+
+	for {
+		keys, resp, err := client.SpacesKeys.List(ctx, opts)
+		if err != nil {
+			return diag.Errorf("Error reading Spaces key: %s", err)
+		}
+
+		for _, k := range keys {
+			if k.AccessKey == d.Id() {
```

Comment:
```suggestion
			if k.Name == d.Get("name").(string) {
```

For the data source, we want something slightly different than the resource. In this case we are looking up an existing key with the given name.